jina/drivers/evaluate.py

```diff
@@ -118,4 +119,36 @@ def _apply_all(self,
                 gt_content = pb2array(gt_content)
             evaluation.value = self.exec_fn(doc_content, gt_content)
             evaluation.op_name = f'{self.metric}-{self.exec.metric}'
-            evaluation.ref_id = groundtruth.id
\ No newline at end of file
+            evaluation.ref_id = groundtruth.id
+
+
+class LoadGroundTruthDriver(KVSearchDriver):
+    """Driver used to search for the `document key` in a KVIndex to find the corresponding groundtruth.
+    (This driver does not use the `recursive structure` of jina Documents, and will not consider the `traversal_path` argument)
+     This driver's job is to fill the `request` groundtruth with the corresponding groundtruth for each document if found in the corresponding KVIndexer.
+    """
+
+    def __init__(self, id_tag: str = 'id', *args, **kwargs):
+        """
+
+        :param id_tag: the name of the tag that corresponds to the key for which to search the `groundtruth` document
+        :param args:
+        :param kwargs:
+        """
+        super().__init__(*args, **kwargs)
+        self.id_tag = id_tag
+
+    def __call__(self, *args, **kwargs):
+        assert len(self.req.groundtruths) == 0
+        miss_idx = []  #: missed hit results, some documents may not have groundtruth and thus will be removed
+        for idx, doc in enumerate(self.req.docs):
+            serialized_groundtruth = self.exec_fn(doc.tags[self.id_tag])
+            if serialized_groundtruth:
+                doc.tags['found'] = True
+                gt = self.req.groundtruths.add()
+                gt.ParseFromString(serialized_groundtruth)
+            else:
+                miss_idx.append(idx)
+            # delete non-existed matches in reverse
+        for j in reversed(miss_idx):
```

Comment:
Is this really wanted? a match, that is not in the ground-truth might be important for evaluation, not?

Reply:
I will clarify this

Reply:
As you can see, it is not using `traversal` so it is only removing `documents` for which there is no `groundtruth`